src/test/java/com/synopsys/integration/jenkins/detect/service/DetectArgumentServiceTest.java

```diff
@@ -0,0 +1,210 @@
+
+        assertEquals(8, detectCommandLine.size()); // Jar args (3) + passed args (2) + auto added args (3)
+        commonValidation(detectCommandLine);
+    }
+
+    @Test
+    public void testVariableReplaced() {
+        // Set up intEnvironmentVariables to only contain custom variable to verify search and replace of the same
+        IntEnvironmentVariables intEnvironmentVariables = new IntEnvironmentVariables(false);
+        intEnvironmentVariables.put("TRUST_CERT", "false");
+        inputDetectProperties.put("--blackduck.trust.cert", "$TRUST_CERT");
+        expectedVisibleDetectProperties.put("--blackduck.trust.cert", "false");
+
+        List<String> detectCommandLine = detectArgumentService
+                                             .parseDetectArgumentString(intEnvironmentVariables, detectJarStrategy.getArgumentEscaper(), detectJarStrategy.getInitialArguments(expectedJavaHome), createDetectPropertiesInputString());
+
+        assertEquals(9, detectCommandLine.size()); // Jar args (3) + passed args (3) + auto added args (3)
+        commonValidation(detectCommandLine);
+    }
+
+    @Test
+    public void testVariableNotReplaced() {
+        // This is used to validate a log entry from calling handleVariableReplacement()
+        inputDetectProperties.put("$VISIBLE", "visible");
+
+        List<String> detectCommandLine = detectArgumentService
+                                             .parseDetectArgumentString(intEnvironmentVariables, detectJarStrategy.getArgumentEscaper(), detectJarStrategy.getInitialArguments(expectedJavaHome), createDetectPropertiesInputString());
+
+        assertEquals(9, detectCommandLine.size()); // Jar args (3) + passed args (3) + auto added args (3)
+        commonValidation(detectCommandLine);
+
+        // Validate log entry from handleVariableReplacement()
+        assertTrue(byteArrayOutputStream.toString().contains("Variable may not have been properly replaced. Argument: $VISIBLE=visible"), "Log should contain message about unable to replace variable.");
+    }
+
+    @Test
+    public void testCustomLogLevel() {
+        inputDetectProperties.put(loggingLevelKey, expectedCustomLogLevel);
+        expectedVisibleDetectProperties.replace(loggingLevelKey, expectedCustomLogLevel);
+
+        List<String> detectCommandLine = detectArgumentService
+                                             .parseDetectArgumentString(intEnvironmentVariables, detectJarStrategy.getArgumentEscaper(), detectJarStrategy.getInitialArguments(expectedJavaHome), createDetectPropertiesInputString());
+
+        assertEquals(8, detectCommandLine.size()); // Jar args (3) + passed args (3) + auto added args (2)
+        commonValidation(detectCommandLine);
+    }
+
+    @Test
+    public void testNoDetectArguments() {
+        expectedVisibleDetectProperties.clear();
```

Comment:
This seems like a strange line. I suspect it'd be better and clearer to pass the exact maps of the hidden and visible properties into commonValidation than for commonValidation to reference specific class-scoped variables as it's doing now.

Reply:
So the standard data within expectedVisibleDetectProperties and expectedHiddenDetectProperties is populated within setUp(). Since this test is testing using no detect arguments, the expectedVisibleDetectProperties needs to be cleared. 

Just seemed unnecessary to pass these Maps into commonValidation, but if it helps in any way to make the tests clearer, I will update that now.